### PR TITLE
NEST-486: Renaming EmailQuotaIndex columns and dropping index

### PR DESCRIPTION
**Pull request by wAsnk**
[NEST-486](https://lombiq.atlassian.net/browse/NEST-486)

[NEST-486]: https://lombiq.atlassian.net/browse/NEST-486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs [hunk 1]
```diff
@@ -25,4 +14,30 @@ public int UpdateFrom1()
 
         return 2;
     }
+
+    public int UpdateFrom2()
+    {
+        // Renaming for safety reasons.
```
Comment (Piedone) [inline Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: What are the safety reasons? Why can't we drop the table regardless of its column names?

File: Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs [hunk 2]
```diff
@@ -1,22 +1,11 @@
...
-using YesSql.Sql;
+using System.Data.Common;
 
 namespace Lombiq.Hosting.Tenants.EmailQuotaManagement.Migrations;
 
 public class EmailQuotaMigrations : DataMigration
 {
-    public int Create()
-    {
-        SchemaBuilder.CreateMapIndexTable<EmailQuotaIndex>(
```
Comment (Piedone) [inline Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs:12]: Delete `EmailQuotaIndex` too.

File: Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs [hunk 3]
```diff
@@ -25,4 +14,30 @@ public int UpdateFrom1()
 
         return 2;
     }
+
+    public int UpdateFrom2()
+    {
+        // Renaming for safety reasons.
+        TryRenameColumn("LastReminder", nameof(EmailQuotaIndex.LastReminderUtc));
```
Comment (Piedone) [inline Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: You can use string names in this class now so `EmailQuotaIndex` can be deleted.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs (hunk 1 above)
Comment (wAsnk) [reply Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: What would happen if I don't rename the column names and then we are looking for the data in the Documents table, but it still has the old column names saved as JSON?

Comment (Piedone) [reply Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: The Document table doesn't reference column names from the index but property names from the `EmailQuota` model.

Comment (wAsnk) [reply Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: Yes, but the EmailQuota model property names were changed as well.

Comment (wAsnk) [reply Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: Hmm, but those might be already saved as the new name I guess. Nvm then.

Comment (Piedone) [reply Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs]: The index and the document might use the same names, but these are still different identifiers.

File: Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs [hunk 4]
```diff
@@ -1,28 +1,23 @@
-using Lombiq.Hosting.Tenants.EmailQuotaManagement.Indexes;
 using OrchardCore.Data.Migration;
-using System;
-using YesSql.Sql;
 
 namespace Lombiq.Hosting.Tenants.EmailQuotaManagement.Migrations;
 
 public class EmailQuotaMigrations : DataMigration
 {
-    public int Create()
```
Comment (Piedone) [inline Lombiq.Hosting.Tenants.EmailQuotaManagement/Migrations/EmailQuotaMigrations.cs:10]: `Create()` needs to remain for new installations but empty and return 3.

**Review by Piedone — APPROVED** [review]

**Pull request merged**